File: C++/weightedRMS.cpp
#include <iostream>

double weightedRMS(double data[], double weights[], int n) {
    double sumOfProducts = 0;
    double sumOfSquaredProducts = 0;
    for(int i = 0; i < n; i++) {
        sumOfProducts += data[i] * weights[i];
        sumOfSquaredProducts = data[i] * data[i] * weights[i];
    }
    return sumOfProducts / sumOfSquaredProducts;
}

File: Python/weightedRMS.py
def weightedRMS(data, weights):

    sumOfProducts = 0
    sumOfSquaredProducts = 0

    for i in range(0, len(data)):
        sumOfProducts += data[i] * weights[i]
        sumOfSquaredProducts = data[i] * data[i] * weights[i]

    return sumOfProducts / sumOfSquaredProducts
